test/controllers/contactPreference/LetterPreferenceControllerSpec.scala

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.contactPreference
+
+import common.SessionKeys
+import controllers.ControllerBaseSpec
+import play.api.http.Status.{BAD_REQUEST, NOT_FOUND, OK, SEE_OTHER}
+import play.api.test.Helpers._
+import forms.YesNoForm.{yes, yesNo, no => _no}
+
+class LetterPreferenceControllerSpec extends ControllerBaseSpec {
+
+  lazy val controller = new LetterPreferenceController(mockErrorHandler)
+
+  "the letterToConfirmedEmailEnabled feature switch is off" when {
+
+
+    ".show() is called" should {
+
+      "return a NOT_FOUND result" in {
+
+        lazy val result = {
+          mockConfig.features.letterToConfirmedEmailEnabled(false)
+          controller.show(request)
+        }
+
+        status(result) shouldBe NOT_FOUND
+      }
+    }
+
+    ".submit() is called" should {
+
+      "return a NOT_FOUND result" in {
+        lazy val result = {
+          mockConfig.features.letterToConfirmedEmailEnabled(false)
+          controller.submit(request)
+        }
+
+        status(result) shouldBe NOT_FOUND
+      }
+    }
+
+  }
+
+  "the letterToConfirmedEmailEnabled feature switch is on" when {
+
+    ".show() is called" should {
+
+      lazy val result = {
+        mockConfig.features.letterToConfirmedEmailEnabled(true)
+        controller.show(request.withSession(SessionKeys.contactPrefUpdate -> "true"))
```

Comment:
the 'withSession' part can be removed as that sessionKey isn't required for this flow